src/libstd/keyword_docs.rs

```diff
@@ -1578,12 +1578,255 @@ mod type_keyword {}
+/// making `unsafe` operations.
+///
+/// See the [Rustnomicon] and the [Reference] for more informations.
+///
+/// # Examples
+///
+/// The following sections are examples of the different uses of `unsafe`, often
+/// with links to go further.
+///
+/// ## Marking elements as `unsafe`
+///
+/// `unsafe` can be used on functions. Note that functions declared in
+/// [`extern`] blocks are implicitly marked as `unsafe`. Methods can also be
+/// declared as `unsafe`:
+///
+/// ```rust
+/// # #![allow(dead_code)]
+/// unsafe fn unsafe_fn() {}
+///
+/// extern "C" {
+///     fn unsafe_extern_fn();
+/// }
+///
+/// trait TraitWithUnsafeMethod {
+///     unsafe fn unsafe_method();
+/// }
+///
+/// struct S;
+///
+/// impl S {
+///     unsafe fn unsafe_method_on_struct() {}
+/// }
+/// ```
+///
+/// Traits can also be declared as `unsafe`:
+///
+/// ```rust
+/// unsafe trait UnsafeTrait {}
+/// ```
+///
+/// Documenting public `unsafe fn` (and other elements) is important. The
+/// standard library has many examples of this, like the following which is an
+/// extract from [`Vec::set_len`]. The `# Safety` section explains the contract
+/// that must be fulfilled to safely call the function. If necessary, a
+/// `# Panics` section can be added to document possible panics and their
+/// reasons.
+///
+/// ```rust,ignore (stub-to-show-doc-example)
+/// /// Forces the length of the vector to `new_len`.
+/// ///
+/// /// This is a low-level operation that maintains none of the normal
+/// /// invariants of the type. Normally changing the length of a vector
+/// /// is done using one of the safe operations instead, such as
+/// /// `truncate`, `resize`, `extend`, or `clear`.
+/// ///
+/// /// # Safety
+/// ///
+/// /// - `new_len` must be less than or equal to `capacity()`.
+/// /// - The elements at `old_len..new_len` must be initialized.
+/// pub unsafe fn set_len(&mut self, new_len: usize)
+/// ```
+///
+/// ## Using `unsafe` blocks and `impl`s
+///
+/// Calling an `unsafe` function requires an `unsafe {}` block, which can take
+/// two forms. Note that the `enclosing_fn` form is the subject of [proposal]
+/// to modify its behavior as an `unsafe {}` block.
+///
+/// ```rust
+/// # #![allow(dead_code)]
+/// unsafe fn unsafe_fn() {}
+///
+/// // This function is unsafe to call and unsafe functions can be called
+/// // inside without problems.
+/// unsafe fn enclosing_unsafe() {
+///     unsafe_fn()
+/// }
 ///
+/// // This function is safe to call. Any call to an unsafe function inside
+/// // must be enclosed in an unsafe block. `interior_unsafe` is responsible
+/// // for ensuring that memory unsafety never occurs, no matter how or when
+/// // it is called.
+/// fn interior_unsafe() {
+///     // SAFETY: `unsafe_fn` does nothing.
+///     unsafe { unsafe_fn() }
+/// }
+/// ```
+///
+/// Traits marked as `unsafe` must be [`impl`]emented using `unsafe`. This makes
+/// a guarantee to other `unsafe` code that a property holds for this type. It
```

Comment:
```suggestion
/// a guarantee to other `unsafe` code that the implementation satisfies the trait's safety contract. It
```
Let's stick to "contract", instead of using "property" for the same thing here.

Reply:
Good catch !